File: redsheet-portal/src/app/database-models/game.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectGameDetail,  vGameRecord  } from './'
export class Game {
    constructor(game?: Game  | vGameRecord )  {
        let properties = [ "gameId", "play", "trigger", "gameTypeId", "projectGameDetailId",];
        for (let property of properties)
            if (game && game[property])
                this[property] = game[property];
    }
 
	gameId : number =   0 ;
 
    @required()
    @maxLength(150)
	play : string =   undefined;
 
    @required()
    @maxLength(150)
	trigger : string =   undefined;
 
    @range(0,2147483647)
	gameTypeId : number =   undefined;
 
    @range(0,2147483647)
	projectGameDetailId : number =   undefined;
	projectGameDetail : ProjectGameDetail  ;


}

File: RedSheet.DbEntities/Models/Game.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Games", Schema = "dbo")]
	[Serializable]
	public partial class Game  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int GameId { get; set; } 
 
	[Required] 
	[MaxLength(150)]
	public string Play { get; set; } 
 
	[Required] 
	[MaxLength(150)]
	public string Trigger { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public GameType GameTypeId { get; set; } 
 
	[RelationshipTableAttribue("ProjectGameDetails","dbo")]
	//Relationships
	public int ProjectGameDetailId { get; set; } 
 
	[ForeignKey("ProjectGameDetailId")]
	public ProjectGameDetail ProjectGameDetail { get; set; } 

	public Game()
	{
    }

    }
}

